utils/sv_IO_panel_tools.py

```diff
@@ -272,11 +272,44 @@ def create_dict_of_tree(ng, skip_set={}, selected=False):
 
         node_dict['params'] = node_items
 
+        # collect custom socket properties
+        print("** PROCESSING custom properties for node: ", node.bl_idname)
+        input_socket_storage = { }
+        for socket in node.inputs:
+
+            # if not tracked_socket(socket): continue
+            print("Socket %d of %d" % (socket.index+1, len(node.inputs)))
+
+            storable = {}
+            tracked_props = 'use_expander', 'use_quicklink', 'expanded', 'use_prop'
+
+            for tracked_prop in tracked_props:
+                if hasattr(socket, tracked_prop):
+                    value = getattr(socket, tracked_prop)
+
+                    print("Processing custom property: ", tracked_prop, " value = ", value)
+
+                    storable[tracked_prop] = value
+
+                    if tracked_prop == 'use_prop' and value:
+                        print("prop type:", type(socket.prop))
+                        storable['prop'] = socket.prop[:]
+                        # storable['socket_prop_value'] = socket.prop[:]
```

Comment:
keep it explicit, verbose even.
``` python
storable['socket_prop_value'] = socket.prop[:] 
```

Reply:
Sure.. but then when you restore from socket_prop_value you have to put that into a prop value. The current implementation actually works. I did use your socket_prop_value initially but the prop value was not set properly. If you prefer an indirection/mapping that’s fine, though I don’t see why we would do that. You are exporting a custom property with the same name as the one stored in the class just like the other props (use_expander etc).